### PR TITLE
add enumerable test ERC721

### PR DESCRIPTION
**Pull request by Geimaj**
We wanted to test transfers though the UI with multiple different contracts so this deploys another ERC721 contract. This contract also implements IERC721Enumerable to enable to UI to display all the tokens owned by the user

### CODE REVIEW

File: ethereum/deploy/12-erc20-app.ts [hunk 1]
```diff
@@ -35,15 +35,21 @@ module.exports = async ({
...
   });
 
   await deployments.deploy("TestToken", {
     from: deployer,
-    args:["Test Token", "TEST"],
+    args: ["Test Token", "TEST"],
+    log: true,
+  });
+
+  await deployments.deploy("TestToken721Enumerable", {
```
Comment (musnit) [inline ethereum/deploy/12-erc20-app.ts:50]: i think this should go in /ethereum/deploy/13-erc721-app.ts rather?

Comment (musnit) [reply ethereum/deploy/12-erc20-app.ts:50]: could you add the loop to mint some nfts to the deployer account too in 13-erc721-app.ts  as well?

Comment (vgeddes) [reply ethereum/deploy/12-erc20-app.ts:50]: I'd rather leave minting and any other custom setup for the dashboard to be done in a separate hardhat script that can be run after the deployments when required. For two reasons:
1) The deployment script already takes a long time to complete, especially with geth.
2) At least for E2E testing purposes, our ERC20 and ERC721 tests are already doing the minting without relying on the deployment script (See https://github.com/Snowfork/snowbridge/blob/main/test/test/erc721app.js#L32).

File: ethereum/scripts/mint-nft.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+const hre = require("hardhat");
```
Comment (vgeddes) [inline ethereum/scripts/mint-nft.ts:1]: Should use ES6 imports:

```suggestion
import { ethers, deployments, getUnnamedAccounts } from "hardhat";
```

**Review by vgeddes — APPROVED** [review]
Looks, good. Added a comment about ES6 imports, but otherwise looks good.

**Review by philipstanislaus — COMMENTED** [review]
LGTM – could you add a bit context what the reason for this addition is? What is it used for?

**Comment by musnit** [comment]
@philipstanislaus ERC721Enumerable advantages here: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC721/extensions/ERC721Enumerable.sol#L9

basically, our frontend can easily query for all tokens owned by a particular user with this, whereas it can't always in non-enumerable tokens. i believe most nft dapps have extra frontend logic on a case-by-case basis to handle non-enumerable tokens, or some other workaround.

**Comment by philipstanislaus** [comment]
Okay, got it. Makes sense for the demo - for a production app we may want to keep the contracts as efficient as possible and use The Graph or another indexing layer for aggregating data.

**Pull request merged**